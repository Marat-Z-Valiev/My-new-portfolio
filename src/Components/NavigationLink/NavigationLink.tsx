import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-scroll";

type NavigationLinkProps = {
	link: string;
	label: string;
	className: string
}

const NavigationLink: React.FC<NavigationLinkProps & {className: string}> = ({
	link,
	label,
}) => {
	return (
		<li className="nav-list-item">
			<Link
				activeClass="active"
				to={link}
				spy={true}
				smooth={true}
				offset={-80}
				duration={500}
			>
				{label}
			</Link>
		</li>
	);
};

export default NavigationLink;
