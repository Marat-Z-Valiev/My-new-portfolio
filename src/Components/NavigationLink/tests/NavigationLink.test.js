import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import NavigationLink from "../NavigationLink";

configure({adapter: new Adapter()});

describe("NavigationLink", () => {
	it("should render correctly", () => {
		const props = {
			activeClass: "active",
			to: "projects",
			spy: true,
			smooth: true,
			offset: -80,
			duration: 500,
		};
		const wrapper = shallow(<NavigationLink {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
