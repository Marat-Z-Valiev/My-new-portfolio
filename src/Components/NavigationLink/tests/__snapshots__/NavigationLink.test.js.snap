// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLink should render correctly 1`] = `
<li
  className="nav-list-item"
>
  <Link
    activeClass="active"
    duration={500}
    offset={-80}
    smooth={true}
    spy={true}
  />
</li>
`;
